modules/index-service/src/main/java/org/opencastproject/index/service/impl/IndexServiceImpl.java

```diff
@@ -2143,4 +2166,40 @@ public boolean isWorkflowActive(String workflowState) {
             || WorkflowState.PAUSED.toString().equals(workflowState);
   }
 
+  private void checkAcceptedTypes(String assetUploadId, MediaType mediaType) {
+    if (mediaType.is(MediaType.OCTET_STREAM)) {
+      // No detailed info, so we have to accept...
+      return;
+    }
+    final JSONParser parser = new JSONParser();
+    try {
+      final Collection<String> assetUploadJsons = listProvidersService.getList("eventUploadAssetOptions",
+          new ResourceListQueryImpl(),false).values();
+      for (String assetUploadJson: assetUploadJsons) {
+        if (!assetUploadJson.startsWith("{") || !assetUploadJson.endsWith("}")) {
+          // Ignore non-json-values
+          continue;
+        }
+        @SuppressWarnings("unchecked")
+        final Map<String, String> assetUpload = (Map<String, String>) parser.parse(assetUploadJson);
```

Comment:
Any idea how fast it is to parse that to JSON every time for every asset?

Reply:
No, but I don't think that this is significant compared to all the other long-running operations which take place when uploading an asset. We are likely to talk about a few milliseconds for the JSON parsing. When I moved it to `RequestUtils` I also declared a static `JSONParser`, so the parser is re-used now.